### PR TITLE
feat: Branch SDK Integration

### PR DESCRIPTION
**Pull request by HamzaIsrar12**
This PR adds the integration of the [Branch SDK](https://branch.io/). The Branch SDK will be used for deep links.

### Config

```
BRANCH:
    ENABLED: false
    KEY: ''
    URI_SCHEME: ''
    HOST: ''
    ALTERNATE_HOST: ''
```

### Reference
- https://help.branch.io/developers-hub/docs/android-basic-integration

### CODE REVIEW

File: app/src/main/java/org/openedx/app/AppActivity.kt [hunk 1]
```diff
@@ -134,6 +138,43 @@ class AppActivity : AppCompatActivity(), InsetHolder, WindowSizeHolder {
         }
     }
 
+    override fun onStart() {
+        super.onStart()
+
+        if (viewModel.isBranchEnabled) {
+            val callback = BranchUniversalReferralInitListener { _, linkProperties, error ->
+                if (linkProperties != null) {
+                    Log.i(BRANCH_TAG, "Branch init complete.")
```
Comment (k1rill) [inline app/src/main/java/org/openedx/app/AppActivity.kt:147]: Can you please use logger from `org.openedx.core.utils`. If there is not enough log levels there, feel free to add them :)

**Review by k1rill — COMMENTED** [review]
Looks good! But one favour, we are still using default_config, can you please add empty branch config fields as an example in current config.yaml files

**Review by k1rill — APPROVED** [review]
LGTM

File: app/build.gradle [hunk 2]
```diff
@@ -139,4 +147,30 @@ dependencies {
     testImplementation "io.mockk:mockk-android:$mockk_version"
     testImplementation "org.jetbrains.kotlinx:kotlinx-coroutines-test:$coroutines_version"
     testImplementation "androidx.arch.core:core-testing:$android_arch_version"
-}
\ No newline at end of file
+}
+
+private def setupBranchConfigFields(buildType) {
```
Comment (omerhabib26) [inline app/build.gradle:152]: We can move this inside ConfigHelper

**Review by omerhabib26 — DISMISSED** [review]
Some minor nits

File: app/build.gradle (hunk 2 above)
Comment (HamzaIsrar12) [reply app/build.gradle:152]: We can handle it in a separate PR.

**Review by omerhabib26 — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -38,6 +45,28 @@
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+
+            <!-- Branch URI Scheme -->
+            <intent-filter>
+                <data
+                    android:host="open"
```
Comment (volodymyr-chekyrta) [inline app/src/main/AndroidManifest.xml:52]: Should we move it to the config as well? 🤔

File: app/src/main/java/org/openedx/app/AppActivity.kt [hunk 4]
```diff
@@ -134,6 +140,43 @@ class AppActivity : AppCompatActivity(), InsetHolder, WindowSizeHolder {
...
+                .init()
+        }
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        this.intent = intent
+
+        if (viewModel.isBranchEnabled) {
+            if (intent?.getBooleanExtra(BRANCH_FORCE_NEW_SESSION, false) == true) {
```
Comment (volodymyr-chekyrta) [inline app/src/main/java/org/openedx/app/AppActivity.kt:168]: How about combining these two into one `if'?

**Pull request merged**